src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -1,21 +1,29 @@
 ﻿using System;
 using System.Collections;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Xml;
-
 using Autodesk.DesignScript.Runtime;
-
 using DSIronPython;
+using Dynamo.Configuration;
 using Dynamo.Graph;
 using Dynamo.Graph.Nodes;
-using ProtoCore.AST.AssociativeAST;
 using Newtonsoft.Json;
-using System.IO;
-using Dynamo.Configuration;
+using Newtonsoft.Json.Converters;
+using ProtoCore.AST.AssociativeAST;
 
 namespace PythonNodeModels
 {
+    /// <summary>
+    /// Enum of possible values of python engine versions
+    /// </summary>
+    public enum PythonEngineVersion
```

Comment:
is it useful for this to be public?

Reply:
I am not sure, I kept @Dewb 's initial scope. Also are internal properties serialized?

Reply:
@mjkkirschner `PythonEngineVersion` cant be less accessible to `PythonNodeBase.Engine`

Reply:
I see, okay. Is this potentially temporary then? - Until we decide if/how we will dynamically add engines?

Reply:
@mjkkirschner I think so, in the end we may move forward to the smart object to determine loaded Pyhthon engine when launching Dynamo. But for now, let's use this for UI features.